compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -921,9 +927,23 @@ TR::TreeTop *TR_OSRLiveRangeAnalysis::collectPendingPush(TR_ByteCodeInfo bci, TR
          int32_t localIndex = local->getLiveLocalIndex();
          _liveVars->set(localIndex);
          if (comp()->getOption(TR_TraceOSR))
-            traceMsg(comp(), "+++ local index %d OSR PPS LIVE\n", localIndex);
+            traceMsg(comp(), "+++ local index %d OSR PENDING PUSH STORE LIVE\n", localIndex);
+         }
+      else if (node->getOpCodeValue() == TR::treetop)
+         {
+         TR::AutomaticSymbol *local = node->getFirstChild()->getSymbolReference()->getSymbol()->getAutoSymbol();
+         int32_t localIndex = local->getLiveLocalIndex();
+         _liveVars->set(localIndex);
+         if (comp()->getOption(TR_TraceOSR))
+            traceMsg(comp(), "+++ local index %d OSR PENDING PUSH LOAD LIVE\n", localIndex);
+
+         if (node->getFirstChild()->getReferenceCount() == 1)
+            {
+            if (comp()->getOption(TR_TraceOSR))
+               traceMsg(comp(), "----> removing node %p\n", node);
+            tt->getPrevTreeTop()->join(tt->getNextTreeTop());
```

Comment:
I know node pools etc are not used really at the moment, but is letting the nodes leak like this correct or is there something else that should be done - I'm not sure. Thoughts @vijaysun-omr ?